azurelinuxagent/ga/cgroupconfigurator.py

```diff
@@ -470,45 +475,70 @@ def disable(self, reason, disable_cgroups):
             except Exception as exception:
                 log_cgroup_warning("Error disabling cgroups: {0}".format(ustr(exception)))
 
-        def _set_cpu_quota(self, unit_name, quota):
-            """
-            Sets CPU quota to the given percentage (100% == 1 CPU)
+        def _get_resource_quotas(self, resource_name):
+            if self._cgroups_api is None:
+                return []
+            if resource_name == ResourceName.CPU:
+                return [CpuQuota(self._cgroups_api)]
+            elif resource_name == ResourceName.MEMORY:
+                return [MemoryQuota(self._cgroups_api)]
+            elif resource_name == ResourceName.ALL:
+                return [CpuQuota(self._cgroups_api), MemoryQuota(self._cgroups_api)]
 
-            NOTE: This is done using a systemtcl set-property --runtime; any local overrides in /etc folder on the VM will take precedence
-            over this setting.
-            """
-            if self._cgroups_api.can_enforce_cpu():
-                quota_percentage = "{0}%".format(quota)
-                log_cgroup_info("Setting {0}'s CPUQuota to {1}".format(unit_name, quota_percentage))
-                CGroupConfigurator._Impl._try_set_cpu_quota(unit_name, quota_percentage)
+            return []
 
-        def _reset_cpu_quota(self, unit_name):
+        def _set_resource_quota(self, unit_name, new_quotas):
+            """
+            Sets the quota for the given resource type ('CPU', 'Memory', or 'All').
+            new_quotas is a dictionary with resource names as keys and quota values as values.
             """
-            Removes any CPUQuota on the agent
+            try:
+                quotas = self._get_resource_quotas(ResourceName.ALL)
+                property_names = []
+                values = []
+                for rq in quotas:
+                    if rq.can_enforce() and rq.name in new_quotas:
+                        q = new_quotas.get(rq.name)
+                        value = rq.format(q)
+                        current = rq.get_current_quota(unit_name)
+                        if current != value:
+                            property_names.append(rq.property)
+                            values.append(value)
+                if property_names:
```

Comment:
Can we be more explicit with this check? 

```suggestion
                if len(property_names) > 0:
```